### ISSUE TITLE
Original vs Keras performance

### ISSUE THREAD

**Issue by rauldiaz** [lyj19940105/pointnet-keras#1]
Hi,

I was wondering why the performance of your implementation is lower than the original repo. Do you have any intuition on why this happens? I also made my own port of pointnet to keras a few months ago and it can't go beyond 82% accuracy on the validation set.

Thanks!

**Comment by garyli1019** [comment]
Hi there, I don't remember what exactly the differences were, but I feel like there are many possible factors could impact the accuracy rate. Like the way we feed the training data(in 2d array, or 3d, or 4d etc), batch number, learning rate decay, optimizer etc. I tried my best to match the original repo, but keras still got some limitations because of its simplicity(I couldn't save my trained model by keras saving feature because of the lambda layer, not sure if they have fixed it or not).

**Comment by oldercoder** [comment]
Hi gary, for save the model, I did save_weights_only in ModelCheckpoint and .save_weights with Model, that works for me. By the way, after the model saved, how do you predict it with a new dataset?

**Comment by Sharknado888** [comment]
Hey guys I found better results using the same code after normalizing the input data into a unit sphere, as was described in the paper.
Thanks garyli1019, for the code.
